src/Symfony/Component/ExpressionLanguage/Tests/ParserTest.php

```diff
@@ -37,6 +37,15 @@ public function testParseWithZeroInNames()
         $parser->parse($lexer->tokenize('foo'), [0]);
     }
 
+    public function testParseUnknownFunction()
+    {
+        $this->expectException(SyntaxError::class);
+        $this->expectExceptionMessage('The function "foo" does not exist around position 1 for expression `foo()`.');
+        $lexer = new Lexer();
+        $parser = new Parser([]);
+        $parser->parse($lexer->tokenize('foo()'));
```

Comment:
should we tokenize before defining the expected exception to ensure only `parse` is allowed to be its source ?